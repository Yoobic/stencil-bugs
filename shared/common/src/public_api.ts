export { CommonModule } from './lib/common.module';

export * from './lib/decorators/mixin/mixin.decorator';
export * from './lib/decorators/persistent/persistent.decorator';
export * from './lib/directives/long-press/long-press.directive';

export * from './lib/interfaces/animate-state/animate-state.class';
export * from './lib/interfaces/animations/animations.class';
export * from './lib/interfaces/colors/colors.interface';
export * from './lib/interfaces/config-constants/config-constants.interface';
export * from './lib/interfaces/position/position.interface';

export * from './lib/pipes/array-from-number/array-from-number.pipe';
export * from './lib/pipes/cloudinary/cloudinary.pipe';
export * from './lib/pipes/date-format/date-format.pipe';
export * from './lib/pipes/file-size/file-size.pipe';
export * from './lib/pipes/highlight/highlight.pipe';
export * from './lib/pipes/https/https.pipe';
export * from './lib/pipes/keys/keys.pipe';
export * from './lib/pipes/limit-to/limit-to.pipe';
export * from './lib/pipes/map-values/map-values.pipe';
export * from './lib/pipes/newline/newline.pipe';
export * from './lib/pipes/nohtml/nohtml.pipe';
export * from './lib/pipes/order-by/order-by.pipe';
export * from './lib/pipes/round/round.pipe';
export * from './lib/pipes/safe/safe.pipe';
export * from './lib/pipes/take/take.pipe';
export * from './lib/pipes/time-ago/time-ago.pipe';
export * from './lib/pipes/timer/timer.pipe';

export * from './lib/services/config-constants/config-constants.service';
export * from './lib/services/core-config/core-config.service';
export * from './lib/services/dom-adapter/dom-adapter.service';
export * from './lib/services/exif-restorer/exif-restorer.service';
export * from './lib/services/geo-location/geo-location.service';
export * from './lib/services/local-forage/local-forage.service';
export * from './lib/services/local-storage/local-storage.service';
export * from './lib/services/log/log.service';
export * from './lib/services/network/network.service';
export * from './lib/services/promise/promise.service';
export * from './lib/services/resize/resize.service';
export * from './lib/services/ruler/ruler.service';
export * from './lib/services/utils/utils.service';

export * from './lib/mixins/lodash-mixins/lodash-mixins';