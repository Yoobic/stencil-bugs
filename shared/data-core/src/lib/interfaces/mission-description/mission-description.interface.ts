import { Translate } from '@shared/translate';
import { IMissionDescription, IScoring, FormFieldType, IFormField  } from '@shared/interfaces';
import { Slide } from '../slide/slide.interface';
import { Model } from '../../decorators/model/model.decorator';
import { Editable } from '../../decorators/editable/editable.decorator';
import { ROLES_CONDITIONS, Condition, getGroupsTransform } from '../condition/condition.interface';
import { Searchable } from '../../decorators/searchable/searchable.decorator';
import { FORM_FILES_IMAGE_FILTER } from '../constants/constants.interface';

export const MISSION_TYPES_NO_ADMIN = ['mission'];
export const MISSION_TYPES = ['mission', 'service', 'poll', 'todo', 'lesson', 'memo']; //'template',

let conditions = {
    isPoll: 'type == "poll"',
    isMission: 'type == "mission"',
    isQuizz: 'quizz == 1',
    isService: 'type == "service"',
    isPollOrService: 'type=="poll" or type=="service"'//,
    //isUpdate: 'isNullOrEmpty(getAttributeValue("_ect")) == 0'
};

export function onMissionDescriptionTypeChange(value, controls, data, field, formDefinition: Array<IFormField>) {
    if (value && (value === 'service' || value === 'poll' || value === 'lesson')) {
        setTimeout(() => {
            controls.language.setValue(Translate.currentLanguage);
        }, 300);
    }
    let groupIndex = formDefinition.findIndex(f => f.name === 'group');
    let group = formDefinition[groupIndex];
    if (value === 'service') {
        group.title = 'SERVICEGROUPS';
    } else {
        group.title = 'MISSIONGROUPS';
    }
    formDefinition[groupIndex] = { ...group };
    return true;
}

@Model({
    className: 'MissionDescription',
    collectionName: 'missiondescriptions',
    fields: [
        'title', 'text', '_id', '_ect', '_lmt', 'icon', 'background', 'group', 'type', 'archived', 'tags', 'missionTags',
        'scoring', 'versionmin', 'skipValidation', 'allowSameUserValidation', 'allowMultiple', 'quizz', 'quizzMode', 'showAnswers',
        'audit', 'recurring', 'category', 'categoryRef', 'language', 'submittext', 'successtext', 'finishedGroups', 'notificationemail', 'pdfMode'
    ],
    include: [] //'category',
})

export class MissionDescription extends IMissionDescription {
    @Editable('MissionDescription', { type: FormFieldType.text, visible: false, forceExport: true, exportOrder: 1 })
    _id?: string;

    @Editable('MissionDescription', {
        type: FormFieldType.autocomplete,
        title: 'TYPE',
        values: MISSION_TYPES_NO_ADMIN,
        translate: true,
        clearable: false,
        required: true,
        exportOrder: 3,
        condition: ROLES_CONDITIONS.isNotTrial,
        onChange: onMissionDescriptionTypeChange
    })
    type: string;

    @Editable('MissionDescription', { required: true, title: 'TITLE', type: FormFieldType.text, exportOrder: 2 })
    @Searchable('MissionDescription')
    title: string;

    @Editable('MissionDescription', { required: true, title: 'DESCRIPTION', type: FormFieldType.textarea, filterable: false, sortable: false, language: 'html' }) //
    text: string;

    @Editable('MissionDescription', { type: FormFieldType.autocomplete, filters: FORM_FILES_IMAGE_FILTER, forceModal: true, hiddenFields: ['mimeType'], mode: 'tile', pageSize: 20, fixedPosition: true, maxWidth: 160, maxHeight: 160, crop: 'circle', collectionName: 'files', title: 'ICON', required: true, flex: 50, columnDefinition: { name: '_downloadURL' }, filterable: false, sortable: false })
    icon: any;

    @Editable('MissionDescription', { type: FormFieldType.autocomplete, filters: FORM_FILES_IMAGE_FILTER, forceModal: true, hiddenFields: ['mimeType'], mode: 'tile', pageSize: 20, fixedPosition: true, maxWidth: 750, maxHeight: 240, crop: 'square', collectionName: 'files', title: 'BACKGROUND', required: true, flex: 50, columnDefinition: { name: '_downloadURL' }, filterable: false, sortable: false })
    background: any;

    @Editable('MissionDescription', { required: true, title: 'MISSIONGROUPS', flex: 100, type: FormFieldType.autocomplete, condition: ROLES_CONDITIONS.isNotTrial, collectionName: 'groups', filters: [[{ field: 'isRole', operator: { _id: 'neq' }, value: true }, { field: 'type', operator: { _id: 'nin' }, value: ['plan'] }]], hiddenFields: ['isRole', 'type'], idOnly: true, mapTransform: getGroupsTransform, multiple: true, clearable: false, exportOrder: 4 })
    //@Editable('MissionDescription', { required: true, title: 'MISSIONGROUPS', flex: 100, type: FormFieldType.autocomplete, condition: ROLES_CONDITIONS.isNotTrial, sessionValues: 'groups', multiple: true, clearable: false, exportOrder: 4 })
    group: Array<string>;

    @Editable('MissionDescription', { required: false, title: 'MISSIONGROUPS', flex: 100, type: FormFieldType.autocomplete, condition: conditions.isService, collectionName: 'groups', filters: [[{ field: 'isRole', operator: { _id: 'neq' }, value: true }, { field: 'type', operator: { _id: 'nin' }, value: ['plan'] }]], hiddenFields: ['isRole', 'type'], idOnly: true, mapTransform: getGroupsTransform, multiple: true, clearable: false, exportOrder: 4 })
    //@Editable('MissionDescription', { required: false, title: 'MISSIONGROUPS', flex: 100, type: FormFieldType.autocomplete, condition: conditions.isService, sessionValues: 'groups', multiple: true, clearable: false })
    serviceGroups: Array<string>;

    @Editable('MissionDescription', { title: 'CAMPAIGNTAGS', type: 'autocomplete', tag: true, collectionName: 'missiondescriptions', multiple: true, subQuery: { field: 'descriptionRef', values: '_id' }, icon: 'yo-flag', condition: ROLES_CONDITIONS.isNotTrial, advanced: true })
    @Searchable('MissionDescription')
    tags: Array<string>;

    @Editable('MissionDescription', { title: 'CATEGORIES', type: 'autocomplete', tag: true, collectionName: 'missions', multiple: true, subQuery: { field: 'descriptionRef', values: '_id' }, icon: 'yo-flag', condition: ROLES_CONDITIONS.isNotTrial, filterable: false })
    @Searchable('MissionDescription')
    missionTags: Array<string>;

    slides: Array<Slide>;

    //@Editable('MissionDescription', { type: FormFieldType.checkbox, title: 'PUBLIC', flex: 100, columnDefinition: { width: 80 }, condition: ROLES_CONDITIONS.isAdmin, filterableAdvanced: true, advanced: true })
    public: boolean;

    @Editable('MissionDescription', { type: FormFieldType.checkbox, title: 'SKIPVALIDATION', condition: ROLES_CONDITIONS.isNotTrial, flex: 100, columnDefinition: { width: 80 }, filterableAdvanced: true, advanced: true })
    skipValidation: boolean;

    @Editable('MissionDescription', { type: FormFieldType.checkbox, title: 'ALLOWSAMEUSERVALIDATION', condition: ROLES_CONDITIONS.isNotTrial, flex: 100, columnDefinition: { width: 80 }, filterableAdvanced: true, advanced: true })
    allowSameUserValidation: boolean;

    @Editable('MissionDescription', { type: FormFieldType.checkbox, condition: conditions.isPoll, flex: 100, columnDefinition: { width: 80 }, filterableAdvanced: true, advanced: true })
    allowMultiple: boolean;

    @Editable('MissionDescription', { type: FormFieldType.checkbox, condition: conditions.isPoll, flex: 100, title: 'QUIZZ', columnDefinition: { width: 80 }, filterableAdvanced: true, advanced: true })
    quizz: boolean;

    @Editable('MissionDescription', { type: FormFieldType.autocomplete, condition: conditions.isQuizz, flex: 100, title: 'MODE', values: ['ALLANSWERSVALID', 'ALLOWUNVALIDANSWERS', 'LIVEANSWERS'], translate: true, columnDefinition: { width: 80 }, filterable: false, advanced: true })
    quizzMode: string;

    @Editable('MissionDescription', { type: FormFieldType.checkbox, condition: conditions.isQuizz, flex: 100, title: 'SHOWANSWERS', columnDefinition: { width: 80 }, filterableAdvanced: true, advanced: true })
    showAnswers: boolean;

    @Editable('MissionDescription', { type: FormFieldType.checkbox, condition: [conditions.isMission, ROLES_CONDITIONS.isNotTrial], flex: 100, title: 'AUDIT', columnDefinition: { width: 80 }, filterableAdvanced: true, advanced: true })
    audit: boolean;

    @Editable('MissionDescription', { type: FormFieldType.checkbox, condition: [conditions.isMission], flex: 100, title: 'RECURRING', columnDefinition: { width: 80 }, filterableAdvanced: true, advanced: true })
    recurring: boolean;
    // @Editable('MissionDescription', { type: FormFieldType.checkbox, flex: 33, title: 'SMALLPHOTOS', columnDefinition: { width: 80 } })
    // smallPhotos: boolean;

    // @Editable('MissionDescription', { type: FormFieldType.autocomplete, title: 'MISSIONCATEGORY', collectionName: 'missioncategories', required: false, columnDefinition: { name: 'title' }, icon: 'yo-list2', filterable: false, sortable: false, condition: ROLES_CONDITIONS.isAdmin, advanced: true })
    category: MissionDescription;

    @Editable('MissionDescription', { type: FormFieldType.autocomplete, required: true, condition: conditions.isPollOrService, title: 'LANGUAGE', flex: 100, translate: true, values: Translate.availablesLanguageAll, clearable: false })
    language: string;

    @Editable('MissionDescription', { condition: conditions.isService, title: 'SUBMITTEXT', flex: 100, type: FormFieldType.text, filterable: false, sortable: false, advanced: true })
    submittext: string;

    @Editable('MissionDescription', { condition: conditions.isService, flex: 100, title: 'SUCCESSTEXT', type: FormFieldType.text, filterable: false, sortable: false, advanced: true })
    successtext: string;

    @Editable('MissionDescription', { title: 'VERSIONMIN', type: FormFieldType.text, sortable: false, condition: ROLES_CONDITIONS.isAdmin, filterableAdvanced: true, advanced: true })
    versionmin: string;

    @Editable('MissionDescription', { required: false, title: 'FINISHEDGROUPS', flex: 100, type: FormFieldType.autocomplete, condition: [ROLES_CONDITIONS.isNotTrial, conditions.isMission], collectionName: 'groups', filters: [[{ field: 'isRole', operator: { _id: 'neq' }, value: true }, { field: 'type', operator: { _id: 'nin' }, value: ['plan'] }]], hiddenFields: ['isRole', 'type'], idOnly: true, mapTransform: getGroupsTransform, multiple: true, clearable: true, exportOrder: 6, advanced: true })
    finishedGroups: Array<string>;

    @Editable('MissionDescription', { title: 'NOTIFICATIONEMAILS', type: FormFieldType.emailreport, showUsers: true, stateful: false, filterable: false, sortable: false, condition: ROLES_CONDITIONS.isNotTrial, advanced: true })
    notificationemail: Array<string>;

    @Editable('MissionDescription', { type: FormFieldType.checkbox, readonly: true, visible: false, forceExport: true, columnDefinition: { width: 80 }, exportOrder: 5, advanced: true })
    archived: boolean;

    @Editable('MissionDescription', { title: 'TOTAL', readonly: true, visible: false, type: FormFieldType.number, sortable: false, filterable: false, advanced: true })
    count: number;

    @Editable('MissionDescription', { readonly: true, visible: false, type: FormFieldType.date, sortable: false, filterable: false, advanced: true })
    latest: Date;

    @Editable('MissionDescription', { title: 'PDF', type: FormFieldType.autocomplete, translate: true, values: ['PDFCLASSIC', 'PDFPHOTOREPORT', 'PDFAUDITEXPORTS', 'PDFCONTRACT'], filterable: false, advanced: true })
    pdfMode?: string;

    @Editable('MissionDescription', { name: '_ect', title: 'CREATIONDATE', type: FormFieldType.datetime, readonly: true, filterableAdvanced: true, advanced: true })
    creationDate: any;

    conditions: Array<Condition>;
    scoring: Array<Scoring>;
}

let scoringConditions = {
    isPercentage: 'isPercentage == 1'
};

@Model({ className: 'Scoring' })
export class Scoring extends IScoring {
    @Editable('Scoring', { required: true, type: FormFieldType.text })
    title: string;

    @Editable('Scoring', { type: FormFieldType.textarea })
    description: string;

    @Editable('Scoring', { title: 'INITIALSCORE', type: FormFieldType.number, advanced: true, condition: ROLES_CONDITIONS.isAdmin })
    initialValue: number;

    @Editable('Scoring', { title: 'MINSCOREFORVALIDATION', type: FormFieldType.number, advanced: true })
    minValue: number;

    @Editable('Scoring', { type: FormFieldType.checkbox, title: 'ISPRIMARYSCORE', columnDefinition: { width: 80 } })
    isActive: boolean;

    @Editable('Scoring', { type: FormFieldType.checkbox, title: 'PERCENTAGE', columnDefinition: { width: 80 }, advanced: true })
    isPercentage: boolean;

    @Editable('Scoring', { title: 'TOTAL', type: FormFieldType.number, condition: scoringConditions.isPercentage, required: true })
    percentageBasis: number;

    @Editable('Scoring', { type: FormFieldType.autocomplete, multiple: true, title: 'FIELDS', displayType: 'formfield' })
    selectedFields: Array<IFormField>;

}
