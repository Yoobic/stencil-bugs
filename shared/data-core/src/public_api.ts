export { DataCoreModule } from './lib/data-core.module';

export * from './lib/decorators/editable/editable.decorator';
export * from './lib/decorators/mapping/mapping.decorator';
export * from './lib/decorators/model/model.decorator';
export * from './lib/decorators/searchable/searchable.decorator';

export * from './lib/interfaces/aggregate-log/aggregate-log.interface';
export * from './lib/interfaces/algorithm/algorithm.interface';
export * from './lib/interfaces/autorenew-config/autorenew-config.interface';
export * from './lib/interfaces/backup/backup.interface';
export * from './lib/interfaces/catalog/catalog.interface';
export * from './lib/interfaces/chart-definition/chart-definition.interface';
export * from './lib/interfaces/condition/condition.interface';
export * from './lib/interfaces/condition/icondition.interface';
export * from './lib/interfaces/constants/constants.interface';
export * from './lib/interfaces/currency/currency.interface';
export * from './lib/interfaces/dashboard/dashboard.interface';
export * from './lib/interfaces/entity/entity.interface';
export * from './lib/interfaces/field-selector/field-selector.interface';
export * from './lib/interfaces/file/file.interface';
export * from './lib/interfaces/folder/folder.interface';
export * from './lib/interfaces/group/group.interface';
export * from './lib/interfaces/kpi/kpi.interface';
export * from './lib/interfaces/location-type/location-type.interface';
export * from './lib/interfaces/location/location.interface';
export * from './lib/interfaces/mapping/mapping.interface';
export * from './lib/interfaces/mission-description/mission-description.interface';
export * from './lib/interfaces/model/model.interface';
export * from './lib/interfaces/notification-event/notification-event.interface';
export * from './lib/interfaces/notification/notification.interface';
export * from './lib/interfaces/operation/operation.interface';
export * from './lib/interfaces/product/product.interface';
export * from './lib/interfaces/response-object/response-object.interface';
export * from './lib/interfaces/slide/slide.interface';
export * from './lib/interfaces/sort/sort.interface';
export * from './lib/interfaces/tableau/tableau.interface';
export * from './lib/interfaces/tag-group/tag-group.interface';
export * from './lib/interfaces/tag/tag.interface';
export * from './lib/interfaces/tenant/tenant.interface';
export * from './lib/interfaces/todo/todo.interface';
export * from './lib/interfaces/translation/translation.interface';
export * from './lib/interfaces/user/user.interface';
export * from './lib/interfaces/workplace/workplace.interface';

export * from './lib/pipes/authorized/authorized.pipe';
export * from './lib/pipes/currency/currency.pipe';
export * from './lib/pipes/filter/filter.pipe';
export * from './lib/pipes/username/username.pipe';
export * from './lib/pipes/useremail/useremail.pipe';
export * from './lib/pipes/userinitial/userinitial.pipe';

export * from './lib/services/activity/activity.service';
export * from './lib/services/algorithms/algorithms.service';
export * from './lib/services/authentication/authentication.guard';
export * from './lib/services/authentication/authentication.service';
export * from './lib/services/authentication/current-session.resolver';
export * from './lib/services/backup/backup.service';
export * from './lib/services/box/box.service';
export * from './lib/services/blog/blog.service';
export * from './lib/services/links/links.service';
export * from './lib/services/broker/broker.service';
export * from './lib/services/config/config.service';
export { Dashboard as DashboardCore } from './lib/services/dashboard/dashboard.service';
export * from './lib/services/data-loader/data-loader.service';
export * from './lib/services/files/files.service';
export * from './lib/services/files/files-broker.service';
export * from './lib/services/googlemaps/googlemaps.service';
export * from './lib/services/googletranslate/googletranslate.service';
export * from './lib/services/http-interceptor/http-interceptor.service';
export * from './lib/services/loading-bar/loading-bar.service';
export * from './lib/services/locations/locations.service';
export * from './lib/services/loopback/loopback.service';
export * from './lib/services/models/models.service';
export * from './lib/services/print/print.service';
export * from './lib/services/push/push.service';
export * from './lib/services/requestor/requestor.service';
export * from './lib/services/security/security.service';
export * from './lib/services/session/session.service';
export * from './lib/services/smartloc/smartloc.service';
export * from './lib/services/users/users.service';
export * from './lib/services/version/version.service';
export * from './lib/services/workplace/workplace.service';
export * from './lib/services/xlsx/xlsx.service';

export * from './lib/validators/password-validator/password-validator';
export * from './lib/validators/username-validator/username-validator';
