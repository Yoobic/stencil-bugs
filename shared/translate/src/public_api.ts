export { TranslateModule } from './lib/translate.module';

export * from './lib/pipes/distance/distance.pipe';
export * from './lib/pipes/filter/filter.pipe';
export * from './lib/pipes/polyglot/polyglot.pipe';

export * from './lib/services/translate/translate.service';