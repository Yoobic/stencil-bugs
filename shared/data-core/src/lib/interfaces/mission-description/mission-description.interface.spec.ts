import { MissionDescription } from './mission-description.interface';
import { Models } from '../../../public_api';

describe('shared: data-core: interfaces: MissionDescription', () => {
    it('should create an entity', () => {
        let entity = new MissionDescription();
        expect(entity._id).toBeUndefined();
        let model = Models.getModel('MissionDescription');
        expect(model).toBeTruthy();
        expect(model).toMatchSnapshot();
    });
});