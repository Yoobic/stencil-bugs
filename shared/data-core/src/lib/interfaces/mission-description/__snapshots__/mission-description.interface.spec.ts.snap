// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared: data-core: interfaces: MissionDescription should create an entity 1`] = `
IModel {
  "_formFields": Array [
    Object {
      "exportOrder": 1,
      "filterName": "_id",
      "forceExport": true,
      "name": "_id",
      "title": "_ID",
      "type": "text",
      "visible": false,
    },
    Object {
      "clearable": false,
      "condition": Object {
        "group": Array [
          "trial",
        ],
        "operator": "nin",
        "type": "roles",
      },
      "exportOrder": 3,
      "filterName": "type",
      "name": "type",
      "onChange": [Function],
      "required": true,
      "title": "TYPE",
      "translate": true,
      "type": "autocomplete",
      "values": Array [
        "mission",
      ],
    },
    Object {
      "exportOrder": 2,
      "filterName": "title",
      "name": "title",
      "required": true,
      "title": "TITLE",
      "type": "text",
    },
    Object {
      "filterName": "text",
      "filterable": false,
      "language": "html",
      "name": "text",
      "required": true,
      "sortable": false,
      "title": "DESCRIPTION",
      "type": "textarea",
    },
    Object {
      "collectionName": "files",
      "columnDefinition": Object {
        "name": "_downloadURL",
      },
      "crop": "circle",
      "filterName": "iconRef",
      "filterable": false,
      "filters": Array [
        Array [
          Object {
            "field": "mimeType",
            "operator": Object {
              "_id": "inq",
            },
            "value": Array [
              "image/png",
              "image/jpg",
              "image/jpeg",
              "image/gif",
              "image/bmp",
            ],
          },
        ],
      ],
      "fixedPosition": true,
      "flex": 50,
      "forceModal": true,
      "hiddenFields": Array [
        "mimeType",
      ],
      "maxHeight": 160,
      "maxWidth": 160,
      "mode": "tile",
      "name": "icon",
      "pageSize": 20,
      "required": true,
      "sortable": false,
      "title": "ICON",
      "type": "autocomplete",
    },
    Object {
      "collectionName": "files",
      "columnDefinition": Object {
        "name": "_downloadURL",
      },
      "crop": "square",
      "filterName": "backgroundRef",
      "filterable": false,
      "filters": Array [
        Array [
          Object {
            "field": "mimeType",
            "operator": Object {
              "_id": "inq",
            },
            "value": Array [
              "image/png",
              "image/jpg",
              "image/jpeg",
              "image/gif",
              "image/bmp",
            ],
          },
        ],
      ],
      "fixedPosition": true,
      "flex": 50,
      "forceModal": true,
      "hiddenFields": Array [
        "mimeType",
      ],
      "maxHeight": 240,
      "maxWidth": 750,
      "mode": "tile",
      "name": "background",
      "pageSize": 20,
      "required": true,
      "sortable": false,
      "title": "BACKGROUND",
      "type": "autocomplete",
    },
    Object {
      "clearable": false,
      "collectionName": "groups",
      "condition": Object {
        "group": Array [
          "trial",
        ],
        "operator": "nin",
        "type": "roles",
      },
      "exportOrder": 4,
      "filterName": "group",
      "filters": Array [
        Array [
          Object {
            "field": "isRole",
            "operator": Object {
              "_id": "neq",
            },
            "value": true,
          },
          Object {
            "field": "type",
            "operator": Object {
              "_id": "nin",
            },
            "value": Array [
              "plan",
            ],
          },
        ],
      ],
      "flex": 100,
      "hiddenFields": Array [
        "isRole",
        "type",
      ],
      "idOnly": true,
      "mapTransform": [Function],
      "multiple": true,
      "name": "group",
      "required": true,
      "title": "MISSIONGROUPS",
      "type": "autocomplete",
    },
    Object {
      "clearable": false,
      "collectionName": "groups",
      "condition": "type == \\"service\\"",
      "exportOrder": 4,
      "filterName": "serviceGroups",
      "filters": Array [
        Array [
          Object {
            "field": "isRole",
            "operator": Object {
              "_id": "neq",
            },
            "value": true,
          },
          Object {
            "field": "type",
            "operator": Object {
              "_id": "nin",
            },
            "value": Array [
              "plan",
            ],
          },
        ],
      ],
      "flex": 100,
      "hiddenFields": Array [
        "isRole",
        "type",
      ],
      "idOnly": true,
      "mapTransform": [Function],
      "multiple": true,
      "name": "serviceGroups",
      "required": false,
      "title": "MISSIONGROUPS",
      "type": "autocomplete",
    },
    Object {
      "advanced": true,
      "collectionName": "missiondescriptions",
      "condition": Object {
        "group": Array [
          "trial",
        ],
        "operator": "nin",
        "type": "roles",
      },
      "filterName": "tags",
      "icon": "yo-flag",
      "multiple": true,
      "name": "tags",
      "subQuery": Object {
        "field": "descriptionRef",
        "values": "_id",
      },
      "tag": true,
      "title": "CAMPAIGNTAGS",
      "type": "autocomplete",
    },
    Object {
      "collectionName": "missions",
      "condition": Object {
        "group": Array [
          "trial",
        ],
        "operator": "nin",
        "type": "roles",
      },
      "filterName": "missionTags",
      "filterable": false,
      "icon": "yo-flag",
      "multiple": true,
      "name": "missionTags",
      "subQuery": Object {
        "field": "descriptionRef",
        "values": "_id",
      },
      "tag": true,
      "title": "CATEGORIES",
      "type": "autocomplete",
    },
    Object {
      "advanced": true,
      "columnDefinition": Object {
        "width": 80,
      },
      "condition": Object {
        "group": Array [
          "trial",
        ],
        "operator": "nin",
        "type": "roles",
      },
      "filterName": "skipValidation",
      "filterableAdvanced": true,
      "flex": 100,
      "name": "skipValidation",
      "title": "SKIPVALIDATION",
      "type": "checkbox",
    },
    Object {
      "advanced": true,
      "columnDefinition": Object {
        "width": 80,
      },
      "condition": Object {
        "group": Array [
          "trial",
        ],
        "operator": "nin",
        "type": "roles",
      },
      "filterName": "allowSameUserValidation",
      "filterableAdvanced": true,
      "flex": 100,
      "name": "allowSameUserValidation",
      "title": "ALLOWSAMEUSERVALIDATION",
      "type": "checkbox",
    },
    Object {
      "advanced": true,
      "columnDefinition": Object {
        "width": 80,
      },
      "condition": "type == \\"poll\\"",
      "filterName": "allowMultiple",
      "filterableAdvanced": true,
      "flex": 100,
      "name": "allowMultiple",
      "title": "ALLOWMULTIPLE",
      "type": "checkbox",
    },
    Object {
      "advanced": true,
      "columnDefinition": Object {
        "width": 80,
      },
      "condition": "type == \\"poll\\"",
      "filterName": "quizz",
      "filterableAdvanced": true,
      "flex": 100,
      "name": "quizz",
      "title": "QUIZZ",
      "type": "checkbox",
    },
    Object {
      "advanced": true,
      "columnDefinition": Object {
        "width": 80,
      },
      "condition": "quizz == 1",
      "filterName": "quizzMode",
      "filterable": false,
      "flex": 100,
      "name": "quizzMode",
      "title": "MODE",
      "translate": true,
      "type": "autocomplete",
      "values": Array [
        "ALLANSWERSVALID",
        "ALLOWUNVALIDANSWERS",
        "LIVEANSWERS",
      ],
    },
    Object {
      "advanced": true,
      "columnDefinition": Object {
        "width": 80,
      },
      "condition": "quizz == 1",
      "filterName": "showAnswers",
      "filterableAdvanced": true,
      "flex": 100,
      "name": "showAnswers",
      "title": "SHOWANSWERS",
      "type": "checkbox",
    },
    Object {
      "advanced": true,
      "columnDefinition": Object {
        "width": 80,
      },
      "condition": Array [
        "type == \\"mission\\"",
        Object {
          "group": Array [
            "trial",
          ],
          "operator": "nin",
          "type": "roles",
        },
      ],
      "filterName": "audit",
      "filterableAdvanced": true,
      "flex": 100,
      "name": "audit",
      "title": "AUDIT",
      "type": "checkbox",
    },
    Object {
      "advanced": true,
      "columnDefinition": Object {
        "width": 80,
      },
      "condition": Array [
        "type == \\"mission\\"",
      ],
      "filterName": "recurring",
      "filterableAdvanced": true,
      "flex": 100,
      "name": "recurring",
      "title": "RECURRING",
      "type": "checkbox",
    },
    Object {
      "clearable": false,
      "condition": "type==\\"poll\\" or type==\\"service\\"",
      "filterName": "language",
      "flex": 100,
      "name": "language",
      "required": true,
      "title": "LANGUAGE",
      "translate": true,
      "type": "autocomplete",
      "values": Array [
        "all",
        "en",
        "us",
        "fr",
        "de",
        "es",
        "nl",
        "pl",
        "it",
        "ru",
        "zhs",
        "zht",
        "pt",
        "kr",
        "ja",
        "ua",
        "he",
        "ar",
        "cz",
        "th",
        "tr",
        "bg",
        "el",
        "sl",
        "sk",
      ],
    },
    Object {
      "advanced": true,
      "condition": "type == \\"service\\"",
      "filterName": "submittext",
      "filterable": false,
      "flex": 100,
      "name": "submittext",
      "sortable": false,
      "title": "SUBMITTEXT",
      "type": "text",
    },
    Object {
      "advanced": true,
      "condition": "type == \\"service\\"",
      "filterName": "successtext",
      "filterable": false,
      "flex": 100,
      "name": "successtext",
      "sortable": false,
      "title": "SUCCESSTEXT",
      "type": "text",
    },
    Object {
      "advanced": true,
      "condition": Object {
        "group": Array [
          "admin",
        ],
        "operator": "in",
        "type": "roles",
      },
      "filterName": "versionmin",
      "filterableAdvanced": true,
      "name": "versionmin",
      "sortable": false,
      "title": "VERSIONMIN",
      "type": "text",
    },
    Object {
      "advanced": true,
      "clearable": true,
      "collectionName": "groups",
      "condition": Array [
        Object {
          "group": Array [
            "trial",
          ],
          "operator": "nin",
          "type": "roles",
        },
        "type == \\"mission\\"",
      ],
      "exportOrder": 6,
      "filterName": "finishedGroups",
      "filters": Array [
        Array [
          Object {
            "field": "isRole",
            "operator": Object {
              "_id": "neq",
            },
            "value": true,
          },
          Object {
            "field": "type",
            "operator": Object {
              "_id": "nin",
            },
            "value": Array [
              "plan",
            ],
          },
        ],
      ],
      "flex": 100,
      "hiddenFields": Array [
        "isRole",
        "type",
      ],
      "idOnly": true,
      "mapTransform": [Function],
      "multiple": true,
      "name": "finishedGroups",
      "required": false,
      "title": "FINISHEDGROUPS",
      "type": "autocomplete",
    },
    Object {
      "advanced": true,
      "condition": Object {
        "group": Array [
          "trial",
        ],
        "operator": "nin",
        "type": "roles",
      },
      "filterName": "notificationemail",
      "filterable": false,
      "name": "notificationemail",
      "showUsers": true,
      "sortable": false,
      "stateful": false,
      "title": "NOTIFICATIONEMAILS",
      "type": "emailreport",
    },
    Object {
      "advanced": true,
      "columnDefinition": Object {
        "width": 80,
      },
      "exportOrder": 5,
      "filterName": "archived",
      "forceExport": true,
      "name": "archived",
      "readonly": true,
      "title": "ARCHIVED",
      "type": "checkbox",
      "visible": false,
    },
    Object {
      "advanced": true,
      "filterName": "count",
      "filterable": false,
      "name": "count",
      "readonly": true,
      "sortable": false,
      "title": "TOTAL",
      "type": "number",
      "visible": false,
    },
    Object {
      "advanced": true,
      "filterName": "latest",
      "filterable": false,
      "name": "latest",
      "readonly": true,
      "sortable": false,
      "title": "LATEST",
      "type": "date",
      "visible": false,
    },
    Object {
      "advanced": true,
      "filterName": "pdfMode",
      "filterable": false,
      "name": "pdfMode",
      "title": "PDF",
      "translate": true,
      "type": "autocomplete",
      "values": Array [
        "PDFCLASSIC",
        "PDFPHOTOREPORT",
        "PDFAUDITEXPORTS",
        "PDFCONTRACT",
      ],
    },
    Object {
      "advanced": true,
      "filterName": "_ect",
      "filterableAdvanced": true,
      "name": "_ect",
      "readonly": true,
      "title": "CREATIONDATE",
      "type": "datetime-local",
    },
  ],
  "appareance": Map {},
  "className": "MissionDescription",
  "collectionName": "missiondescriptions",
  "feathersService": undefined,
  "fields": Array [
    "title",
    "text",
    "_id",
    "_ect",
    "_lmt",
    "icon",
    "background",
    "group",
    "type",
    "archived",
    "tags",
    "missionTags",
    "scoring",
    "versionmin",
    "skipValidation",
    "allowSameUserValidation",
    "allowMultiple",
    "quizz",
    "quizzMode",
    "showAnswers",
    "audit",
    "recurring",
    "category",
    "categoryRef",
    "language",
    "submittext",
    "successtext",
    "finishedGroups",
    "notificationemail",
    "pdfMode",
  ],
  "include": Array [],
  "isCustom": undefined,
  "mappingFields": Map {},
  "searchSubquery": undefined,
  "searchableFields": Array [
    "title",
    "tags",
    "missionTags",
  ],
  "type": [Function],
}
`;
